### ISSUE TITLE
java.lang.RuntimeException when attempting screenshot. / StringUtils.compareStrings(StringUtils.java:266)

### ISSUE THREAD

**Issue by tonymontgomery** [onesky/auto-screenshot-android#12]
Similar to existing issue:
https://github.com/onesky/auto-screenshot-android/issues/11

I am unable to capture anything.
Using v0.7.3.

Do we have access to the OneSkyScreenshotHelper & StringUtils code?
If so I can see what is happening at line 266 of StringUtils.

Error:
D/OneSkyScreenshotHelper(31767): Capturing ListView (8)
D/OneSkyScreenshotHelper(31767): Capturing LinearLayout (2)
D/OneSkyScreenshotHelper(31767): Found instance of TextView with text: Search
D/OneSkyScreenshotHelper(31767): Comparing text with all string resources, text: Search
E/Fatal   (31767): java.lang.RuntimeException: Error receiving broadcast Intent { act=com.oneskyapp.screenshot.ScreenshotButtonService.CAPTURE_ACTION flg=0x10 } in com.oneskyapp.screenshot.OneSkyScreenshotHelper$1@18651de1
E/Fatal   (31767): 	at com.oneskyapp.screenshot.utils.StringUtils.compareStrings(StringUtils.java:266)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:364)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.getScreenshot(OneSkyScreenshotHelper.java:255)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.capture(OneSkyScreenshotHelper.java:231)
E/Fatal   (31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper$1.onReceive(OneSkyScreenshotHelper.java:116)
W/System.err(31767): java.lang.RuntimeException: Error receiving broadcast Intent { act=com.oneskyapp.screenshot.ScreenshotButtonService.CAPTURE_ACTION flg=0x10 } in com.oneskyapp.screenshot.OneSkyScreenshotHelper$1@18651de1
W/System.err(31767): 	at com.oneskyapp.screenshot.utils.StringUtils.compareStrings(StringUtils.java:266)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:364)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.captureLocalizableStrings(OneSkyScreenshotHelper.java:345)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.getScreenshot(OneSkyScreenshotHelper.java:255)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper.capture(OneSkyScreenshotHelper.java:231)
W/System.err(31767): 	at com.oneskyapp.screenshot.OneSkyScreenshotHelper$1.onReceive(OneSkyScreenshotHelper.java:116)

**Comment by tonymontgomery** [comment]
For what it's worth... I tried an older version (v 0.7.2) and I have finally got it to work.
From here--> https://github.com/onesky/auto-screenshot-android/releases

However, the camera button is **VERY** hard to press. I have to press it maybe 50 times before it will finally register as a press. (no exaggeration) 

I will open a different issue for the button press

**Comment by bretdabaker** [comment]
at StringUtils.java:266 we compare the displayed text with all localized text using regex.

The log you posted does not provide information on what is causing the Runtime exception, it could be a NullPointerException or PatternSyntaxException. Do you have a more detail log?

Or would you mind to share you strings.xml so that I can debug?

**Comment by tonymontgomery** [comment]
```
I added print statements in the StringUtils code and I believe the issue is in one of the strings below.
Perhaps the escaping of percent sign? (%%)
Not sure. Like stated above, StringUtils in v0.7.2 works so I just used that for now.
If you don't find an issue with the strings below let me know... I can send you my strings.xml file. I would rather email it to you than share with everybody.

Thanks!

-Tony

    <string name="promotion_campaign_popup">You have %1$d days remaining to claim your special offer of 30%% off a second product.</string>
    <string name="promotion_congratz">Congratulations on your ACME purchase!</string>
    <string name="promotion_hint">Tap here to track your progress…</string>
    <string name="promotion_info">Get 30%% back from ACME when you connect a second ACME speaker in %1$d Days. Tap Learn More for details.</string>
    <string name="promotion_info_spotify">Get 30%% back from ACME when you connect a second ACME speaker in %1$d Days. Tap Learn More for details and to receive Premium for %2$d Days.</string>
    <string name="promotion_info_simple">Amazing Rewards Await You…</string>
    <string name="promotion_offer_popup_first">Congratulations on purchasing your second ACME device! You have %1$d days to redeem your rebate.</string>
    <string name="promotion_offer_popup_next">You have %1$d days to redeem your rebate.</string>
```

**Comment by bretdabaker** [comment]
Fixed in [v0.7.4](https://github.com/onesky/auto-screenshot-android/releases/tag/v0.7.4)

**Comment by tonymontgomery** [comment]
Verified fixed in v0.7.4

Thanks!
